pkg/profile/artifact.go

```diff
@@ -94,6 +103,61 @@ func (pa *ProfilesArtifactsMaker) Make(installation profilesv1.ProfileInstallati
 	return pa.generateOutput(filepath.Join(profileRootdir, "profile-installation.yaml"), &installation)
 }
 
+// makeArtifacts creates artifacts. This is a separate function from the main Make function in order to handle
+// nested profiles recursively.
+func (pa *ProfilesArtifactsMaker) makeArtifacts(installation profilesv1.ProfileInstallation) ([]Artifact, error) {
+	path := installation.Spec.Source.Path
+	branchOrTag := installation.Spec.Source.Tag
+	if installation.Spec.Source.Tag == "" {
+		branchOrTag = installation.Spec.Source.Branch
+	}
+	definition, err := getProfileDefinition(installation.Spec.Source.URL, branchOrTag, path, pa.GitClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get profile definition: %w", err)
+	}
+	var artifacts []Artifact
+
+	for _, artifact := range definition.Spec.Artifacts {
+		if pa.nestedName != "" {
+			artifact.Name = filepath.Join(pa.nestedName, artifact.Name)
+		}
+
+		if err := validateArtifact(artifact); err != nil {
+			return nil, fmt.Errorf("validation failed for artifact %s: %w", artifact.Name, err)
+		}
+
+		t := -1
+		if artifact.Profile != nil {
+			profileRepoName := profileRepo(installation)
+			if containsKey(pa.profileRepos, profileRepoName) {
+				return nil, fmt.Errorf("recursive artifact detected: profile %s on branch %s contains an artifact that points recursively back at itself", artifact.Profile.Source.URL, artifact.Profile.Source.Branch)
+			}
+			pa.profileRepos = append(pa.profileRepos, profileRepoName)
+			nestedArtifacts, err := pa.handleNestedProfile(artifact, installation)
+			if err != nil {
+				return nil, err
+			}
+			artifacts = append(artifacts, nestedArtifacts...)
+			pa.nestedName = ""
+			pa.profileRepos = nil
+			continue
+		} else if artifact.Kustomize != nil {
+			t = KUSTOMIZE
+		} else if artifact.Chart != nil {
+			t = CHART
+		}
+		if t == -1 {
+			return nil, errors.New("no artifact set")
+		}
+		arts, err := pa.Builders[t].Build(artifact, installation, definition)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build artifact: %w", err)
+		}
```

Comment:
I find `t` super confusing, wdyt about this refactor:
```suggestion
		} else if artifact.Kustomize != nil {
			builder = pa.Builders[KUSTOMIZE]
		} else if artifact.Chart != nil {
			builder = pa.Builders[CHART]
		} else {
			return nil, errors.New("no artifact set")
		}
		arts, err := builder.Build(artifact, installation, definition)
		if err != nil {
			return nil, fmt.Errorf("failed to build artifact: %w", err)
		}
```